google/cloud/aiplatform/training_jobs.py

```diff
@@ -358,9 +358,16 @@ def state(self) -> Optional[gca_pipeline_state.PipelineState]:
 
         return self._gca_resource.state
 
-    def get_model(self) -> Optional[models.Model]:
+    @base.optional_sync()
```

Comment:
Is it as simple as adding this annotation?

@sasha-gitg

Reply:
The flow should look something like this:

```python

def get_model(self, sync=True) -> Optional[models.Model]:
  self._assert_has_run()
  if not self._gca_resource.model_to_upload:
    raise RuntimeError(self._model_upload_fail_string)

  return self._force_get_model(sync=sync)

@base.optional_sync()
def _force_get_model(self, sync=True) -> models.Model:
  model = self._get_model()
  if model is None:
    raise RuntimeError(self._model_upload_fail_string)
  return model
```

This will deal with the scenario where the user calls job.get_model(sync=False) on a job that does not produce a Model.

You can add these changes and preferably a test or to unblock this PR, this can be removed for now and can you open a ticket to add aync support to get_model and add as a TODO (please add this comment to the ticket for posterity if you do).

Reply:
Added a test